game/scripts/core/Weapon.gd

```diff
@@ -1,67 +1,70 @@
-extends Node
+extends Node2D
 class_name Weapon
 
-@export var force: float
-@export var damage: float
-@export var stops_movment: bool 
-@export var player_force: float
-# Need something here to denote when you can equip it. 
-@export var equipable_slots: int
-@onready var held_sprite: Sprite2D = $HeldSprite
-@onready var use_sprite: Sprite2D = $UseSprite
-@onready var UseAnimation: AnimationPlayer = $UseAnimation
-@onready var held_hitbox: HitBox = $HeldHitbox
-@onready var use_hitbox: HitBox = $Marker2D/UseHitbox
-var movement_stopped: bool = false
+@export var weapon_data: WeaponData
+@export var combo_window: float = 0.2  # Seconds
+@onready var active_attack_index: int = 0
+@onready var attacks: Array[Attack]
+@onready var attacking: bool = false
+@onready var combo_timer = $ComboTimer
+@export var attack_scenes: Array[PackedScene] = []
 
-# Called when the node enters the scene tree for the first time.
-func _ready():
-	use_sprite.visible = false
-	use_hitbox.disable()
+		
+func _increment_attack_index():
+	active_attack_index = (active_attack_index + 1) % len(attacks)
 
+func _create_attacks():
+	# Helper function to attach attacks to a weapon given 
+	# the supplied packed scenes
+	for attack_index in weapon_data.attack_indices:
+		var attack: Attack = attack_scenes[attack_index].instantiate()
+		add_child(attack)
+		attack.attack_apply_hit_signal.connect(apply_hit)
+		attack.attack_end_signal.connect(attack_end)
+		attack.attack_start_signal.connect(attack_start)
+		attack.disable()
+		attacks.append(attack)
+
+
+func _ready():
+	assert (weapon_data.attack_indices.max() < len(attack_scenes))
+	_create_attacks()
+	
+	
+func flip_h():
+	# Horizontally flip the whole weapon.
+	scale = Vector2(scale[0] * -1, 1)
+	
+func use():
+	if not attacking:
+		attacks[active_attack_index].use()
+		_increment_attack_index()
 
-# Called every frame. 'delta' is the elapsed time since the previous frame.
-func _process(delta):
-	pass
-#
-#func use():
-#	UseAnimation.play("use")
-#
 
 func equip(equipment_slot):
```

Comment:
Wonder if we will need this here. We can discuss how to connect these componentes inventory, player, weapon